boot/reboot.go

```diff
@@ -57,9 +57,22 @@ var (
 	shutdownMsg = i18n.G("reboot scheduled to update the system")
 	haltMsg     = i18n.G("system halt scheduled")
 	poweroffMsg = i18n.G("system poweroff scheduled")
+
+	// testingRebootItself is set to true when we want to unit
+	// test the Reboot function.
+	testingRebootItself = false
 )
 
+func EnableTestingRebootFunction() (restore func()) {
+	testingRebootItself = true
+	return func() { testingRebootItself = false }
+}
+
 func Reboot(action RebootAction, rebootDelay time.Duration, rebootInfo *RebootInfo) error {
+	if osutil.IsTestBinary() && !testingRebootItself {
+		return nil
```

Comment:
usually we would panic here, see initramfsReboot in initramfs.go for example

Reply:
Changed now

Reply:
not pushed perhaps, I still see "return nil"

Reply:
Oh, right - should be good now